src/libs/UserUtils.ts

```diff
@@ -83,22 +100,35 @@ function getDefaultAvatar(accountID = -1): React.FC<SvgProps> {
 
     // There are 24 possible default avatars, so we choose which one this user has based
     // on a simple modulo operation of their login number. Note that Avatar count starts at 1.
-    const accountIDHashBucket = ((accountID % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
 
+    // When creating a chat, we generate an avatar using an ID and the backend response will modify the ID to the actual user ID.
+    // But the avatar link still corresponds to the original ID-generated link. So we extract the SVG image number from the backend's link instead of using the user ID directly
+    let accountIDHashBucket: AvatarRange;
+    if (avatarURL) {
+        const match = avatarURL.match(/(?<=default-avatar_)\d+(?=\.)/);
```

Comment:
we have one more default avatar prefix  `avatar_` which is missed here, that caused a regression more details [here](https://github.com/Expensify/App/issues/33452#issuecomment-1867016966)